src/make_round_config.R

```diff
@@ -1 +1,162 @@
-print("Once implemented, script will update hub-config/tasks.config based on latest list of clades to model.")
\ No newline at end of file
+## Add a new round to the hub's task config (hub-config/tasks.json)
+## This script is part of a "round open" process run via a scheduled GitHub action and
+## should be run from the hub's src/ directory.
+
+library(hubAdmin)
+library(hubUtils)
+library(tools)
+
+get_clade_file_info <- function(hub_dir) {
+  # the round_id is the date of the most recent "clades to model"
+  # file in auxiliary-data/modeled-clades
+  directory <- file.path(hub_dir, "auxiliary-data", "modeled-clades")
+  clades_file_list <- sort(list.files(path = directory, full.names = TRUE), decreasing = TRUE)
+  latest_clades_file <- clades_file_list[1]
+  clade_file_info <- list(clade_file = latest_clades_file, round_id = file_path_sans_ext(basename(latest_clades_file)))
+  return(clade_file_info)
+}
+
+get_clade_list <- function(filename) {
+  # get a list of the clades to be modeled in this round
+  clade_list <- readLines(filename)
+  return(clade_list)
+}
+
+create_new_round <- function() {
+  # create a new round object
+  this_round_clade_file <- get_clade_file_info(dirname(getwd()))
+  this_round_clade_list <- sort(get_clade_list(this_round_clade_file$clade_file))
+  this_round_date <- this_round_clade_file$round_id
+  if (isFALSE(weekdays(as.Date(this_round_date)) == "Wednesday")) {
+    stop("The latest clade_file does not have a Wednesday date: ", this_round_clade_file)
+  }
+
+  model_tasks <- hubAdmin::create_model_tasks(
+    hubAdmin::create_model_task(
+      task_ids = hubAdmin::create_task_ids(
+        hubAdmin::create_task_id("nowcast_date",
+          required = list(this_round_date),
+          optional = NULL
+        ),
+        hubAdmin::create_task_id("target_date",
+          required = NULL,
+          ## target date is nowcast_date and the three prior weeks
+          optional = as.character(as.Date(this_round_date) - c(0, 7, 14, 21))
+        ),
+        hubAdmin::create_task_id("location",
+          required = NULL,
+          optional = c(
+            "AL", "AK", "AZ", "AR", "CA", "CO",
+            "CT", "DE", "DC", "FL", "GA", "HI",
+            "ID", "IL", "IN", "IA", "KS", "KY",
+            "LA", "ME", "MD", "MA", "MI", "MN",
+            "MS", "MO", "MT", "NE", "NV", "NH",
+            "NJ", "NM", "NY", "NC", "ND", "OH",
+            "OK", "OR", "PA", "RI", "SC", "SD",
+            "TN", "TX", "UT", "VT", "VA", "WA",
+            "WV", "WI", "WY", "PR"
+          )
+        ),
+        hubAdmin::create_task_id("variant",
+          required = this_round_clade_list,
+          optional = NULL
+        )
+      ),
+      output_type = hubAdmin::create_output_type(
```

Comment:
Should this script include another output type besides sample?

Reply:
Yes, we want samples to be optional and "mean" output type to be required.